docs/server-side-search.rst

```diff
@@ -64,3 +64,107 @@ and then click on :guilabel:`Search Analytics`.
    Search analytics demo
 
 .. _Elasticsearch: https://www.elastic.co/products/elasticsearch
+
+API
+---
+
+Search is exposed through our API that's proxied from the domain where your docs are being served.
+This is ``https://docs.readthedocs.io/_/api/v2/docsearch`` for the ``docs`` project, for example.
```

Comment:
I wonder if we should just name this `_/api/v2/search`? Would that be too annoying on the code side for backwards compat?

Reply:
Actually I like that name, but if we don't need to maintain both for a long time... I mean, leave the old endpoint till we have migrated the other pieces and then remove it, I'm +1 on that.